tools/generator/GenBaseSupport.cs

```diff
@@ -302,12 +302,14 @@ public override bool ValidateNamespace ()
 			return true;
 		}
 
-		bool IsObfuscatedName (String name)
+		bool IsObfuscatedName (int threshold, string name)
 		{
 			if (name.StartsWith ("R.", StringComparison.Ordinal))
 				return false;
 			int idx = name.LastIndexOf ('.');
 			string last = idx < 0 ? name : name.Substring (idx + 1);
+			if (threshold <= Math.Pow (26, last.Length - 1))
```

Comment:
I'm stupid at math. What I wanted was, always treat one-letter name as obfuscation result, treat two-letter names as obfuscated only when there are 26+ classes in the package, same for three-letter names only when there are 26^2+ classes, and so on... Therefore the condition should be:

    (!(threshold >= Math.Pow (26, length - 1))

Note that this condition is to filter out non-obfuscated case ("return false").

I avoided simple length <= 2 because there can be FooBar.id or FooBar.in (possibly with FooBar.out). IMO that's no-go.

But you're right, that calculating Math.Pow() for lengthy string doesn't make sense (moreover, there's possible overflow...!). It should be trimmed down to, maybe 3? There wouldn't be 17576 types in a package, even if the library contains a lot of automatically generated types (I'm hesitant to set threshold at 100 or 1000 because of possibly generated types in a single package, 1000 is possible).

(The calculation above then covers "id" ? No, but when there are aa, ab, ... ic, id, ie etc. then this "id" wouldn't be distinct. At least we can say we assumed so).